pkg/config/config.go

```diff
@@ -322,108 +327,99 @@ func NewConfig(nn types.NamespacedName, uid types.UID, currentState *SpiceDBMigr
 	}, warning, nil
 }
 
-func validateImage(image string, globalConfig *OperatorConfig) (string, []error) {
-	if len(image) == 0 && len(globalConfig.ImageName) == 0 {
-		return "", []error{fmt.Errorf("no defualt image configured for operator and no image provided in spec")}
+func computeTargets(image, version, channel, engine string, currentVersion *v1alpha1.SpiceDBVersion, globalConfig *OperatorConfig, rolling bool) (baseImage string, targetSpiceDBVersion *v1alpha1.SpiceDBVersion, state updates.State, err error) {
```

Comment:
maybe the {image, version, channel} should be its own struct being passed around?

Reply:
this is pretty much the only place we need all 3 I think? we mostly use version/channel